src/core/thread/data_poll_manager.hpp

```diff
@@ -225,8 +241,9 @@ class DataPollManager : public InstanceLocator
     uint32_t    GetDefaultPollPeriod(void) const;
 
     uint32_t mTimerStartTime;
-    uint32_t mExternalPollPeriod;
     uint32_t mPollPeriod;
+    uint32_t mExternalPollPeriod : 26; //< In milliseconds, about 18.64 hours ((1 << 26) / (1000 * 3600)) in maximum.
```

Comment:
Good idea on changing this to use bit-field :+1: 

How about updating `SetExternalPollPeriod()` to ensure its parameter `uint32_t aPeriod` is capped to fit it the `mExternalPollPeriod` (if `aPreiod` is larger than max, we use max instead)? Also updating documentation to mention this?